### PR TITLE
Sn 22 posts api endpoint

### PR DESCRIPTION
**Pull request by mlshvch**

### CODE REVIEW

File: app/controllers/api/v1/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      end
+
+      # GET /posts/:id or /posts/:id.json
+      def show
+        render json: @post, status: :ok
+      end
+
+      # POST /posts or /posts.json
+      def create
+        if post_params.blank?
```
Comment (PFDeve) [inline app/controllers/api/v1/posts_controller.rb]: we may move these(in `create` and `update` methods) conditions to `before_action`

File: app/controllers/api/v1/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            render json: @post, status: :created
+          else
+            render json: @post.errors, status: :unprocessable_entity
+          end
+        end
+      end
+
+      # PUT or PATCH /posts/:id or /posts/:id.json
+      def update
+        if post_params.blank?
```
Comment (PFDeve) [inline app/controllers/api/v1/posts_controller.rb]: I mean this too

**Review by PFDeve — COMMENTED** [review]
I noted the fat conditions in another PR 
- https://github.com/iTechArt-RubyLab/social-network/pull/28
- https://github.com/iTechArt-RubyLab/social-network/pull/24
- https://github.com/iTechArt-RubyLab/social-network/pull/30

If we move these, it'll be more readeble and simplify the logic

File: app/controllers/api/v1/posts_controller.rb (hunk 1 above)
Comment (ramantechart) [reply app/controllers/api/v1/posts_controller.rb]: We can get rid of this completely.
Anyway, we validate `if post.save`

File: app/controllers/api/v1/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      end
+
+      private
+
+      def set_post
+        @post = Post.find(params[:id])
+      end
+
+      def post_params
+        params.permit(
```
Comment (ramantechart) [inline app/controllers/api/v1/posts_controller.rb]: missing require

File: app/serializers/post_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+# class responsible for serializing profile data
+class PostSerializer < ActiveModel::Serializer
```
Comment (ramantechart) [inline app/serializers/post_serializer.rb:4]: keep in mind we already have one more PostSerializer in another PR

File: app/serializers/post_serializer.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  attributes :id, :body, :status, :created_at, :updated_at, :user_id
```
Comment (ramantechart) [inline app/serializers/post_serializer.rb]: missing picture

**Review by ramantechart — CHANGES_REQUESTED** [review]

**Review by skripchenkoVeta — APPROVED** [review]

**Pull request merged**